plugins/wp-graphql-headless-webhooks/src/WebhookRegistry.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Webhook Registry class
+ *
+ * @package WPGraphQL\Webhooks
+ */
+
+namespace WPGraphQL\Webhooks;
+
+/**
+ * Class WebhookRegistry
+ */
+class WebhookRegistry {
+
+	/**
+	 * Registered webhook types
+	 *
+	 * @var array<string, array<string, mixed>> Array of webhook types keyed by type identifier.
+	 */
+	private array $webhook_types = [];
+
+	/**
+	 * Instance of the registry
+	 *
+	 * @var WebhookRegistry|null
+	 */
+	private static ?WebhookRegistry $instance = null;
+
+	/**
+	 * Get registry instance
+	 *
+	 * @return WebhookRegistry
+	 */
+	public static function instance(): WebhookRegistry {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initialize the registry
+	 */
+	public static function init(): void {
+		// Register the CPT on initialization.
+		add_action( 'init', [ self::instance(), 'register_webhook_cpt' ], 5, 0 );
+		do_action( 'graphql_register_webhooks', self::instance() );
+	}
+
+	/**
+	 * Register the webhook CPT
+	 */
+	public function register_webhook_cpt(): void {
```

Comment:
Maybe there is a benefit to have Post registration functionality separately, in a separate class? 

Also can we discuss what benefits we have when utilizing it as a Post Type vs Options/Settings?

Reply:
Thanks @whoami-pwd . I will refactor. After discussion with @josephfusco  I think using CPT provides several benefics over settings:

CPTs provide built-in admin UI, revision history, meta fields per webhook, and native REST/GraphQL integration. We can create multiple hooks as a separate database entry rather than bloating a single serialized options array. The CPT structure also enables more granular permissions as well. So Im not sure how scalable the Options/Settings variant will be in that case.